lib/Conversion/TorchToTMTensor/TorchToTMTensor.cpp

```diff
@@ -552,6 +568,270 @@ class ConvertAtenMaxPool2dWithIndicesBackwardOp
+              loc, rewriter.getIntegerAttr(srcType.getElementType(), 1));
+        }
+      } else if (reduceType == "amax") {
+        // Set the values in the input tensor to the smallest element of that type
+        auto minAttr = getNumericLimit(rewriter, srcType.getElementType(), /*getMin=*/true);
+        normalizationValue = rewriter.create<arith::ConstantOp>(loc, minAttr);
+      } else if (reduceType == "amin") {
+        // Set the values in the input tensor to the largest element of that type
+        auto maxAttr = getNumericLimit(rewriter, srcType.getElementType(), /*getMin=*/false);
+        normalizationValue = rewriter.create<arith::ConstantOp>(loc, maxAttr);
+      }
+      Value normalizations = createInitTensor(
+          rewriter, loc, SmallVector<Value>({indexSize}),
+          srcType.getElementType(), /*init_element*/ normalizationValue);
+      self = createTMTensorScatterOp(
+          rewriter, loc, normalizations, indices, self,
+          /*uniqueIndices=*/false,
+          [&](OpBuilder &b, Location loc, Value update, Value current) {
+            b.create<TMTensor::YieldOp>(loc, update);
+      });
+      if (reduceType == "mean") {
+          counts = createTMTensorScatterOp(
+              rewriter, loc, normalizations, indices, counts,
+              /*uniqueIndices=*/false,
+              [&](OpBuilder &b, Location loc, Value update, Value current) {
+                b.create<TMTensor::YieldOp>(loc, update);
+              });
+      }
+    }
+
+    // Create final operation
+    Value scatterOp = createTMTensorScatterOp(
+        rewriter, loc, updates, indices, self,
+        /*uniqueIndices=*/false,
+        [&](OpBuilder &b, Location loc, Value update, Value current) {
+          Value result;
+          if (reduceType == "sum" || reduceType == "mean") {
+            if (update.getType().isa<mlir::IntegerType>()) {
+              result = b.create<arith::AddIOp>(loc, update, current);
+            } else {
+              result = b.create<arith::AddFOp>(loc, update, current);
+            }
+          } else if (reduceType == "prod") {
+            if (update.getType().isa<mlir::IntegerType>()) {
+              result = b.create<arith::MulIOp>(loc, update, current);
+            } else {
+              result = b.create<arith::MulFOp>(loc, update, current);
+            }
+          } else if (reduceType == "amax") {
+            if (update.getType().isa<mlir::IntegerType>()) {
+              result = b.create<arith::MaxSIOp>(loc, update, current);
+            } else {
+              result = b.create<arith::MaxFOp>(loc, update, current);
+            }
+          } else if (reduceType == "amin") {
+            if (update.getType().isa<mlir::IntegerType>()) {
+              result = b.create<arith::MinSIOp>(loc, update, current);
+            } else {
+              result = b.create<arith::MinFOp>(loc, update, current);
+            }
+          }
```

Comment:
nit: add an else with an `llvm_unreachable` to make it explicit that every case is being handled

Reply:
Added this to all cases